src/Language/Bitcoin/Script/Descriptors/Syntax.hs

```diff
@@ -2,6 +2,8 @@ module Language.Bitcoin.Script.Descriptors.Syntax (
     OutputDescriptor (..),
     ScriptDescriptor (..),
     KeyDescriptor (..),
+    ChecksumDescriptor (..),
```

Comment:
I think it would make more sense to define `ChecksumDescriptor` and `ChecksumStatus` in the `Parser` module (or maybe in `Checksum`).